### ISSUE TITLE
Strange internal compiler error: compiler/rustc_infer/src/infer/region_constraints/mod.rs:568:17: cannot relate bound region

### ISSUE THREAD

**Issue by minus199** [rust-lang/rust#110988]
### Code

```Rust
async fn extract_auth(response: Response) -> Result<Auth, String> {
    let cookies: HashMap<String, String> = HashMap::new();
    response
        .cookies()
        .into_iter()
        .fold(cookies, |mut acc, cookie| {
            acc.insert(cookie.name().to_string(), cookie.value().to_string());
            acc
        });

// The error occurs when we uncomment this code. not sure if relevant but the fix is the code above
    // let mut cookies: HashMap<String, String> =
    //     response
    //         .cookies()
    //         .fold(cookies, |acc: &mut HashMap<String, String>, cookie:Cookie| {
    //             acc.insert(cookie.name().to_string(), cookie.value().to_string());
    //             acc
    //         });

    let text = response.text().await.unwrap();
    let req_verficiation_token =
        extract_verification_token(text.as_str()).expect("unable to fetch token");

    let auth = Auth {
        request_verification_token: "todo!()",
        uzma: "todo!()",
        uzmb: "todo!()",
        uzmc: "todo!()",
        uzmd: "todo!()",
        uzme: "todo!()",
        session_id_mypost: "todo!()",
    };

    Ok(auth)
}
```

### Meta

`rustc --version --verbose`:
```
rustc 1.68.2 (9eb3afe9e 2023-03-27)
binary: rustc
commit-hash: 9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0
commit-date: 2023-03-27
host: x86_64-unknown-linux-gnu
release: 1.68.2
LLVM version: 15.0.6

```

### Error output

```
error: internal compiler error: compiler/rustc_infer/src/infer/region_constraints/mod.rs:568:17: cannot relate bound region: ReLateBound(DebruijnIndex(0), BoundRegion { var: 1, kind: BrNamed(DefId(0:416 ~ tracker[cc77]::opx::extract_auth::'_#1), '_) }) <= '_#1r

thread 'rustc' panicked at 'Box<dyn Any>', /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/compiler/rustc_errors/src/lib.rs:987:33
stack backtrace:
   0:     0x7f9ea599959a - std::backtrace_rs::backtrace::libunwind::trace::ha271a8a7e1f3d4ef
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f9ea599959a - std::backtrace_rs::backtrace::trace_unsynchronized::h85739da0352c791a
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f9ea599959a - std::sys_common::backtrace::_print_fmt::hbc6ebcfb2910b329
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x7f9ea599959a - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::he1c117e52d53614f
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f9ea59fb39e - core::fmt::write::h25eb51b9526b8e0c
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/fmt/mod.rs:1213:17
   5:     0x7f9ea5989be5 - std::io::Write::write_fmt::ha9edec5fb1621933
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/io/mod.rs:1682:15
   6:     0x7f9ea5999365 - std::sys_common::backtrace::_print::hf8657cd429fc3452
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:47:5
   7:     0x7f9ea5999365 - std::sys_common::backtrace::print::h41b9b18ed86f86bd
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:34:9
   8:     0x7f9ea599c12f - std::panicking::default_hook::{{closure}}::h22a91871f4454152
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:267:22
   9:     0x7f9ea599be6b - std::panicking::default_hook::h21ddc36de0cd4ae7
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:286:9
  10:     0x7f9ea8cec324 - rustc_driver[70f63b52fde826b7]::DEFAULT_HOOK::{closure#0}::{closure#0}
  11:     0x7f9ea599c96a - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h6f7e3c94ecc52e2f
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/alloc/src/boxed.rs:2002:9
  12:     0x7f9ea599c96a - std::panicking::rust_panic_with_hook::h5059419d6d59b3d0
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:692:13
  13:     0x7f9ea8fa7f81 - std[f8b79e573431a86c]::panicking::begin_panic::<rustc_errors[53e74ce803854a4d]::ExplicitBug>::{closure#0}
  14:     0x7f9ea8fa08d6 - std[f8b79e573431a86c]::sys_common::backtrace::__rust_end_short_backtrace::<std[f8b79e573431a86c]::panicking::begin_panic<rustc_errors[53e74ce803854a4d]::ExplicitBug>::{closure#0}, !>
  15:     0x7f9ea9030da6 - std[f8b79e573431a86c]::panicking::begin_panic::<rustc_errors[53e74ce803854a4d]::ExplicitBug>
  16:     0x7f9ea9030d96 - std[f8b79e573431a86c]::panic::panic_any::<rustc_errors[53e74ce803854a4d]::ExplicitBug>
  17:     0x7f9ea902f6c2 - <rustc_errors[53e74ce803854a4d]::HandlerInner>::span_bug::<rustc_span[7c23fb27ec020b97]::span_encoding::Span, &alloc[b346c7f99e9347e5]::string::String>
  18:     0x7f9ea902f567 - <rustc_errors[53e74ce803854a4d]::Handler>::span_bug::<rustc_span[7c23fb27ec020b97]::span_encoding::Span, &alloc[b346c7f99e9347e5]::string::String>
  19:     0x7f9ea8fe4acb - rustc_middle[6dfce017f6b7786d]::util::bug::opt_span_bug_fmt::<rustc_span[7c23fb27ec020b97]::span_encoding::Span>::{closure#0}
  20:     0x7f9ea8fe4b1a - rustc_middle[6dfce017f6b7786d]::ty::context::tls::with_opt::<rustc_middle[6dfce017f6b7786d]::util::bug::opt_span_bug_fmt<rustc_span[7c23fb27ec020b97]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  21:     0x7f9ea8fe3bda - rustc_middle[6dfce017f6b7786d]::ty::context::tls::with_context_opt::<rustc_middle[6dfce017f6b7786d]::ty::context::tls::with_opt<rustc_middle[6dfce017f6b7786d]::util::bug::opt_span_bug_fmt<rustc_span[7c23fb27ec020b97]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  22:     0x7f9ea8fe3b06 - rustc_middle[6dfce017f6b7786d]::util::bug::opt_span_bug_fmt::<rustc_span[7c23fb27ec020b97]::span_encoding::Span>
  23:     0x7f9ea8fe3ac4 - rustc_middle[6dfce017f6b7786d]::util::bug::span_bug_fmt::<rustc_span[7c23fb27ec020b97]::span_encoding::Span>
  24:     0x7f9ea6c77604 - <rustc_middle[6dfce017f6b7786d]::ty::sty::Region as rustc_middle[6dfce017f6b7786d]::ty::relate::Relate>::relate::<rustc_infer[83c7bda8c7ffae41]::infer::equate::Equate>
  25:     0x7f9ea6c728bf - rustc_middle[6dfce017f6b7786d]::ty::relate::super_relate_tys::<rustc_infer[83c7bda8c7ffae41]::infer::equate::Equate>
  26:     0x7f9ea6c6f8be - <rustc_infer[83c7bda8c7ffae41]::infer::equate::Equate as rustc_middle[6dfce017f6b7786d]::ty::relate::TypeRelation>::tys
  27:     0x7f9ea7501db3 - <rustc_infer[83c7bda8c7ffae41]::infer::InferCtxt>::commit_if_ok::<rustc_infer[83c7bda8c7ffae41]::infer::InferOk<()>, rustc_middle[6dfce017f6b7786d]::ty::error::TypeError, <rustc_infer[83c7bda8c7ffae41]::infer::at::Trace>::eq<rustc_middle[6dfce017f6b7786d]::ty::Ty>::{closure#0}>
  28:     0x7f9ea7501ca6 - <rustc_infer[83c7bda8c7ffae41]::infer::at::At>::eq::<rustc_middle[6dfce017f6b7786d]::ty::Ty>
  29:     0x7f9ea98ca39e - <rustc_infer[83c7bda8c7ffae41]::infer::InferCtxt>::can_eq::<rustc_middle[6dfce017f6b7786d]::ty::Ty>
  30:     0x7f9ea98ad1d2 - rustc_trait_selection[d385e741675330b5]::traits::error_reporting::suggestions::hint_missing_borrow
  31:     0x7f9ea9950934 - <rustc_infer[83c7bda8c7ffae41]::infer::error_reporting::TypeErrCtxt as rustc_trait_selection[d385e741675330b5]::traits::error_reporting::suggestions::TypeErrCtxtExt>::report_closure_arg_mismatch
  32:     0x7f9ea9960788 - <rustc_infer[83c7bda8c7ffae41]::infer::error_reporting::TypeErrCtxt as rustc_trait_selection[d385e741675330b5]::traits::error_reporting::TypeErrCtxtExt>::report_selection_error
  33:     0x7f9ea996a167 - <rustc_infer[83c7bda8c7ffae41]::infer::error_reporting::TypeErrCtxt as rustc_trait_selection[d385e741675330b5]::traits::error_reporting::InferCtxtPrivExt>::report_fulfillment_error
  34:     0x7f9ea995d7cf - <rustc_infer[83c7bda8c7ffae41]::infer::error_reporting::TypeErrCtxt as rustc_trait_selection[d385e741675330b5]::traits::error_reporting::TypeErrCtxtExt>::report_fulfillment_errors
  35:     0x7f9ea80007ce - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_call
  36:     0x7f9ea6e788b2 - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  37:     0x7f9ea6edf5cf - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_match::{closure#0}
  38:     0x7f9ea6e7ad6e - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  39:     0x7f9ea6e78d87 - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  40:     0x7f9ea6eb166e - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_decl
  41:     0x7f9ea6eae773 - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_block_with_expected
  42:     0x7f9ea6e78d3c - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  43:     0x7f9ea6e7b26f - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  44:     0x7f9ea6eae8c1 - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_block_with_expected
  45:     0x7f9ea6e78d3c - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  46:     0x7f9ea6e138d1 - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_return_expr
  47:     0x7f9ea6e0a802 - rustc_hir_typeck[698e9809084ff9ec]::check::check_fn
  48:     0x7f9ea7eb11c1 - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_expr_closure
  49:     0x7f9ea6e7ce0b - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  50:     0x7f9ea6eee90d - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_argument_types
  51:     0x7f9ea7ffa98b - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_call
  52:     0x7f9ea6e788b2 - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  53:     0x7f9ea6e138d1 - <rustc_hir_typeck[698e9809084ff9ec]::fn_ctxt::FnCtxt>::check_return_expr
  54:     0x7f9ea6e0a802 - rustc_hir_typeck[698e9809084ff9ec]::check::check_fn
  55:     0x7f9ea6df30be - rustc_hir_typeck[698e9809084ff9ec]::typeck
  56:     0x7f9ea6de4797 - <rustc_query_system[59807289ca97d521]::dep_graph::graph::DepGraph<rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_span[7c23fb27ec020b97]::def_id::LocalDefId, &rustc_middle[6dfce017f6b7786d]::ty::typeck_results::TypeckResults>
  57:     0x7f9ea6ddc2eb - rustc_query_system[59807289ca97d521]::query::plumbing::try_execute_query::<rustc_query_impl[81153385010d7886]::queries::typeck, rustc_query_impl[81153385010d7886]::plumbing::QueryCtxt>
  58:     0x7f9ea86a4e74 - <rustc_query_impl[81153385010d7886]::Queries as rustc_middle[6dfce017f6b7786d]::ty::query::QueryEngine>::typeck
  59:     0x7f9ea7d7c057 - rustc_mir_build[fb6c695b6e0f4a7e]::thir::cx::thir_body
  60:     0x7f9ea7d7e60c - <rustc_query_system[59807289ca97d521]::dep_graph::graph::DepGraph<rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_middle[6dfce017f6b7786d]::ty::WithOptConstParam<rustc_span[7c23fb27ec020b97]::def_id::LocalDefId>, core[1d432356d8e1e9f1]::result::Result<(&rustc_data_structures[560608da72c35763]::steal::Steal<rustc_middle[6dfce017f6b7786d]::thir::Thir>, rustc_middle[6dfce017f6b7786d]::thir::ExprId), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>>
  61:     0x7f9ea7d72b96 - rustc_query_system[59807289ca97d521]::query::plumbing::try_execute_query::<rustc_query_impl[81153385010d7886]::queries::thir_body, rustc_query_impl[81153385010d7886]::plumbing::QueryCtxt>
  62:     0x7f9ea7d6b6f4 - rustc_mir_build[fb6c695b6e0f4a7e]::build::mir_built
  63:     0x7f9ea7a1ebbb - <rustc_query_system[59807289ca97d521]::dep_graph::graph::DepGraph<rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_middle[6dfce017f6b7786d]::ty::WithOptConstParam<rustc_span[7c23fb27ec020b97]::def_id::LocalDefId>, &rustc_data_structures[560608da72c35763]::steal::Steal<rustc_middle[6dfce017f6b7786d]::mir::Body>>
  64:     0x7f9ea7a1dda0 - rustc_query_system[59807289ca97d521]::query::plumbing::try_execute_query::<rustc_query_impl[81153385010d7886]::queries::mir_built, rustc_query_impl[81153385010d7886]::plumbing::QueryCtxt>
  65:     0x7f9ea74a34b6 - rustc_mir_transform[2d65a293f14beadb]::check_unsafety::unsafety_check_result
  66:     0x7f9ea74a2ec4 - <rustc_mir_transform[2d65a293f14beadb]::check_unsafety::provide::{closure#0} as core[1d432356d8e1e9f1]::ops::function::FnOnce<(rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_span[7c23fb27ec020b97]::def_id::LocalDefId)>>::call_once
  67:     0x7f9ea7bec3c7 - <rustc_query_system[59807289ca97d521]::dep_graph::graph::DepGraph<rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_span[7c23fb27ec020b97]::def_id::LocalDefId, &rustc_middle[6dfce017f6b7786d]::mir::query::UnsafetyCheckResult>
  68:     0x7f9ea7be9297 - rustc_query_system[59807289ca97d521]::query::plumbing::try_execute_query::<rustc_query_impl[81153385010d7886]::queries::unsafety_check_result, rustc_query_impl[81153385010d7886]::plumbing::QueryCtxt>
  69:     0x7f9ea7be4eaa - rustc_mir_transform[2d65a293f14beadb]::mir_const
  70:     0x7f9ea7a1ebbb - <rustc_query_system[59807289ca97d521]::dep_graph::graph::DepGraph<rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_middle[6dfce017f6b7786d]::ty::WithOptConstParam<rustc_span[7c23fb27ec020b97]::def_id::LocalDefId>, &rustc_data_structures[560608da72c35763]::steal::Steal<rustc_middle[6dfce017f6b7786d]::mir::Body>>
  71:     0x7f9ea7be3de8 - rustc_query_system[59807289ca97d521]::query::plumbing::try_execute_query::<rustc_query_impl[81153385010d7886]::queries::mir_const, rustc_query_impl[81153385010d7886]::plumbing::QueryCtxt>
  72:     0x7f9ea7bdb31b - rustc_mir_transform[2d65a293f14beadb]::mir_promoted
  73:     0x7f9ea7bdaa7e - <rustc_query_system[59807289ca97d521]::dep_graph::graph::DepGraph<rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_middle[6dfce017f6b7786d]::ty::WithOptConstParam<rustc_span[7c23fb27ec020b97]::def_id::LocalDefId>, (&rustc_data_structures[560608da72c35763]::steal::Steal<rustc_middle[6dfce017f6b7786d]::mir::Body>, &rustc_data_structures[560608da72c35763]::steal::Steal<rustc_index[525e17cad9ac453c]::vec::IndexVec<rustc_middle[6dfce017f6b7786d]::mir::Promoted, rustc_middle[6dfce017f6b7786d]::mir::Body>>)>
  74:     0x7f9ea7bd991a - rustc_query_system[59807289ca97d521]::query::plumbing::try_execute_query::<rustc_query_impl[81153385010d7886]::queries::mir_promoted, rustc_query_impl[81153385010d7886]::plumbing::QueryCtxt>
  75:     0x7f9ea7bd7b58 - rustc_borrowck[aa2c7e496ca54303]::mir_borrowck
  76:     0x7f9ea7bd7574 - <rustc_borrowck[aa2c7e496ca54303]::provide::{closure#0} as core[1d432356d8e1e9f1]::ops::function::FnOnce<(rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_span[7c23fb27ec020b97]::def_id::LocalDefId)>>::call_once
  77:     0x7f9ea7542897 - <rustc_query_system[59807289ca97d521]::dep_graph::graph::DepGraph<rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_span[7c23fb27ec020b97]::def_id::LocalDefId, &rustc_middle[6dfce017f6b7786d]::mir::query::BorrowCheckResult>
  78:     0x7f9ea75415f7 - rustc_query_system[59807289ca97d521]::query::plumbing::try_execute_query::<rustc_query_impl[81153385010d7886]::queries::mir_borrowck, rustc_query_impl[81153385010d7886]::plumbing::QueryCtxt>
  79:     0x7f9ea86a521d - <rustc_query_impl[81153385010d7886]::Queries as rustc_middle[6dfce017f6b7786d]::ty::query::QueryEngine>::mir_borrowck
  80:     0x7f9ea84cb2ad - rustc_hir_analysis[651175ae4d1746c1]::collect::type_of::type_of
  81:     0x7f9ea74f7e96 - <rustc_query_system[59807289ca97d521]::dep_graph::graph::DepGraph<rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_span[7c23fb27ec020b97]::def_id::DefId, rustc_middle[6dfce017f6b7786d]::ty::Ty>
  82:     0x7f9ea74f575a - rustc_query_system[59807289ca97d521]::query::plumbing::get_query::<rustc_query_impl[81153385010d7886]::queries::type_of, rustc_query_impl[81153385010d7886]::plumbing::QueryCtxt, rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>
  83:     0x7f9ea784c69c - rustc_hir_analysis[651175ae4d1746c1]::check::check::check_mod_item_types
  84:     0x7f9ea750ec67 - <rustc_query_system[59807289ca97d521]::dep_graph::graph::DepGraph<rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, rustc_span[7c23fb27ec020b97]::def_id::LocalDefId, ()>
  85:     0x7f9ea825fde3 - rustc_query_system[59807289ca97d521]::query::plumbing::try_execute_query::<rustc_query_impl[81153385010d7886]::queries::check_mod_item_types, rustc_query_impl[81153385010d7886]::plumbing::QueryCtxt>
  86:     0x7f9ea86a477d - <rustc_query_impl[81153385010d7886]::Queries as rustc_middle[6dfce017f6b7786d]::ty::query::QueryEngine>::check_mod_item_types
  87:     0x7f9ea7181b27 - <rustc_session[cd7c78827427c6d6]::session::Session>::time::<(), rustc_hir_analysis[651175ae4d1746c1]::check_crate::{closure#6}>
  88:     0x7f9ea7181332 - rustc_hir_analysis[651175ae4d1746c1]::check_crate
  89:     0x7f9ea7180fab - rustc_interface[698bad460dbd68d3]::passes::analysis
  90:     0x7f9ea8417f76 - <rustc_query_system[59807289ca97d521]::dep_graph::graph::DepGraph<rustc_middle[6dfce017f6b7786d]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[6dfce017f6b7786d]::ty::context::TyCtxt, (), core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>>
  91:     0x7f9ea8417131 - rustc_query_system[59807289ca97d521]::query::plumbing::try_execute_query::<rustc_query_impl[81153385010d7886]::queries::analysis, rustc_query_impl[81153385010d7886]::plumbing::QueryCtxt>
  92:     0x7f9ea86a160a - <rustc_query_impl[81153385010d7886]::Queries as rustc_middle[6dfce017f6b7786d]::ty::query::QueryEngine>::analysis
  93:     0x7f9ea7ecd010 - <rustc_interface[698bad460dbd68d3]::passes::QueryContext>::enter::<rustc_driver[70f63b52fde826b7]::run_compiler::{closure#1}::{closure#2}::{closure#2}, core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>>
  94:     0x7f9ea7ecaa84 - rustc_span[7c23fb27ec020b97]::with_source_map::<core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>, rustc_interface[698bad460dbd68d3]::interface::run_compiler<core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>, rustc_driver[70f63b52fde826b7]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  95:     0x7f9ea7ec32d4 - <scoped_tls[1044df3f3db9be03]::ScopedKey<rustc_span[7c23fb27ec020b97]::SessionGlobals>>::set::<rustc_interface[698bad460dbd68d3]::interface::run_compiler<core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>, rustc_driver[70f63b52fde826b7]::run_compiler::{closure#1}>::{closure#0}, core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>>
  96:     0x7f9ea7ec29d2 - std[f8b79e573431a86c]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[698bad460dbd68d3]::util::run_in_thread_pool_with_globals<rustc_interface[698bad460dbd68d3]::interface::run_compiler<core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>, rustc_driver[70f63b52fde826b7]::run_compiler::{closure#1}>::{closure#0}, core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>>
  97:     0x7f9ea7ec277a - <<std[f8b79e573431a86c]::thread::Builder>::spawn_unchecked_<rustc_interface[698bad460dbd68d3]::util::run_in_thread_pool_with_globals<rustc_interface[698bad460dbd68d3]::interface::run_compiler<core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>, rustc_driver[70f63b52fde826b7]::run_compiler::{closure#1}>::{closure#0}, core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[1d432356d8e1e9f1]::result::Result<(), rustc_errors[53e74ce803854a4d]::ErrorGuaranteed>>::{closure#1} as core[1d432356d8e1e9f1]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  98:     0x7f9ea59a6823 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h3205ec2d7fc232c5
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/alloc/src/boxed.rs:1988:9
  99:     0x7f9ea59a6823 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h3bb5daec8177f56b
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/alloc/src/boxed.rs:1988:9
 100:     0x7f9ea59a6823 - std::sys::unix::thread::Thread::new::thread_start::had7b8061e306bb5c
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys/unix/thread.rs:108:17
 101:     0x7f9ea56dcb43 - start_thread
                               at ./nptl/./nptl/pthread_create.c:442:8
 102:     0x7f9ea576ea00 - clone3
                               at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
 103:                0x0 - <unknown>

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.68.2 (9eb3afe9e 2023-03-27) running on x86_64-unknown-linux-gnu

note: compiler flags: --crate-type bin -C embed-bitcode=no -C debuginfo=2 -C incremental=[REDACTED]

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
#0 [typeck] type-checking `opx::extract_auth`
#1 [thir_body] building THIR for `opx::extract_auth`
#2 [mir_built] building MIR for `opx::extract_auth`
#3 [unsafety_check_result] unsafety-checking `opx::extract_auth`
#4 [mir_const] preparing `opx::extract_auth` for borrow checking
#5 [mir_promoted] processing MIR for `opx::extract_auth`
#6 [mir_borrowck] borrow-checking `opx::extract_auth`
#7 [type_of] computing type of `opx::extract_auth::{opaque#0}`
#8 [check_mod_item_types] checking item types in module `opx`
#9 [analysis] running analysis passes on this crate
end of query stack
warning: `tracker` (bin "tracker") generated 13 warnings
error: could not compile `tracker`; 13 warnings emitted

```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
<backtrace>
```

</p>
</details>

**Comment by Nilstrieb** [comment]
Thank you for the report! This is a duplicate of #109361 and fixed in 1.70.0, so I'm closing it as such.

**Comment by jyn514** [comment]
Hmm, did we not backport the fix to 1.69?

**Comment by Nilstrieb** [comment]
Ah yeah we did, I was confused by that issue still not being closed. I will close it.
